### PR TITLE
[docs] Update deepspeed docs, add some more information and link to streamlit

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

Updates the DeepSpeed docs with reference to single file checkpointing when using ZeRO 3 + some helpful information.

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/advanced/advanced_gpu.rst [hunk 1]
```diff
@@ -516,6 +525,27 @@ Here is some helpful information when setting up DeepSpeed ZeRO Stage 3 with Lig
 * When offloading to the CPU, make sure to bump up the batch size as GPU memory will be freed
 * We also support sharded checkpointing. By passing ``save_full_weights=False`` to the ``DeepSpeedPlugin``, we'll save shards of the model which allows you to save extremely large models. However to load the model and run test/validation/predict you must use the Trainer object.
 
+.. _deepspeed-zero-stage-3-single-file:
+
+Collating Single File Checkpoint for DeepSpeed ZeRO Stage 3
+"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+
```
Comment (awaelchli) [inline docs/source/advanced/advanced_gpu.rst:533]:
```suggestion
"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""

```

File: docs/source/advanced/advanced_gpu.rst [hunk 2]
```diff
@@ -1,8 +1,10 @@
-Advanced GPU Optimized Training
-===============================
+Advanced GPU Optimized Training & Model Parallelism
+===================================================
 
 When training large models, fitting larger batch sizes, or trying to increase throughput using multi-GPU compute, Lightning provides advanced optimized distributed training plugins to support these cases and offer substantial improvements in memory usage.
 
+In many cases these plugins are some flavour of model parallelism however we only introduce concepts at a high level to get you started. Refer to the `FairScale documentation <https://fairscale.readthedocs.io/en/latest/deep_dive/oss_sdp_fsdp.html>`__ to for more information about model parallelism.
+
```
Comment (awaelchli) [inline docs/source/advanced/advanced_gpu.rst:7]:
```suggestion
In many cases these plugins are some flavour of model parallelism however we only introduce concepts at a high level to get you started. Refer to the `FairScale documentation <https://fairscale.readthedocs.io/en/latest/deep_dive/oss_sdp_fsdp.html>`__  for more information about model parallelism.

```

File: docs/source/advanced/advanced_gpu.rst [hunk 3]
```diff
@@ -516,6 +525,27 @@ Here is some helpful information when setting up DeepSpeed ZeRO Stage 3 with Lig
...
+    # enter directory where you saved your checkpoints
+    cd lightning_logs/version_0/checkpoints/
+    # lightning has saved a directory instead of a file
+    cd epoch=0-step=0.ckpt/
+
+    python convert_lightning_deepspeed_checkpoint.py . pytorch_model.ckpt
+
+.. warning::
+
+    This single file checkpoint does not include the optimizer/lr states. This means we cannot restore training via `resume_from_checkpoint`. Ensure to keep the sharded checkpoint directory if this is required.
```
Comment (awaelchli) [inline docs/source/advanced/advanced_gpu.rst:547]:
```suggestion
    This single file checkpoint does not include the optimizer/lr-scheduler states. This means we cannot restore training via the `resume_from_checkpoint` Trainer argument. Ensure to keep the sharded checkpoint directory if this is required.
```

**Review by awaelchli — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

File: docs/source/advanced/advanced_gpu.rst [hunk 4]
```diff
@@ -516,6 +525,26 @@ Here is some helpful information when setting up DeepSpeed ZeRO Stage 3 with Lig
...
+After training using ZeRO Stage 3, you'll notice that your checkpoints are a directory of sharded model and optimizer states. If you'd like to collate a single file from the checkpoint directory please use `this custom gist <https://gist.github.com/SeanNaren/e0535977df8e165247f7ff9ae445be1b>`__ which handles all the Lightning states additionally when collating the file.
+
+.. code-block:: bash
+
+    # enter directory where you saved your checkpoints
+    cd lightning_logs/version_0/checkpoints/
+    # lightning has saved a directory instead of a file
+    cd epoch=0-step=0.ckpt/
+
+    python convert_lightning_deepspeed_checkpoint.py . pytorch_model.ckpt
```
Comment (kaushikb11) [inline docs/source/advanced/advanced_gpu.rst:542]: Should we introduce this script functionality in Lightning?

**Review by kaushikb11 — APPROVED** [review]

File: docs/source/advanced/advanced_gpu.rst (hunk 4 above)
Comment (SeanNaren) [reply docs/source/advanced/advanced_gpu.rst:542]: Great idea, lemme make a separate PR!

**Pull request merged**